### PR TITLE
Asset : Added Test For AssetInSameState

### LINKED ISSUE #381: [C4GT] Asset: Add tests for `AssetInSameState`
## Description
Subtask under : https://github.com/cord-network/community/issues/7

New testcase can be tested under `cargo test -p pallet-asset` after adding it in the code.

## Goals

- [ ] Add tests for `AssetInSameState` for pallet/asset

## Expected Outcome
- [ ] Test should assert for `AssetInSameState` being returned properly in all the possible calls.
- [ ] If there are more than one function returning this error code, all of them should be validated either as one test case or as multiple test cases.
       
## Acceptance Criteria
NA

## Implementation Details
Look at other test cases, and add a test case for the same.
Ex: check method `asset_create_should_succeed` which validates creation of asset.

## Mockups / Wireframes
NA

---- 

## Product Name
CORD

## Organization Name
Dhiway

## Domain
Blockchain

## Tech Skills Needed
Rust

## Mentor(s)
@amarts @vatsa287 

## Complexity
[Low]

## Category
[Test]

## Sub Category
[Beginner friendly]

### PR DESCRIPTION
**Pull request by zeel991**
- Fixes https://github.com/dhiway/cord/issues/381
- In the test , I have covered both the cases where the we could get `AssetInSameState`
- Please Review the PR and give your feedback

### CODE REVIEW

**Comment by zeel991** [comment]
Hey @amarts @vatsa287 , the PR has passed all the three tests

**Review by vatsa287 — CHANGES_REQUESTED** [review]
Please check for both the instances
- When instance-id is provided and asset is in same state,
- When instance-id is not provided, then change status of asset itself, but the given new status is of same status of existing asset.

Above must be done for both status_change & vc_status_change.
So totally 4 places to check the error-code `AssetInSameState`.

**Comment by zeel991** [comment]
Hey @vatsa287 , I hope I have completed your required changes 
Please review the PR

File: pallets/asset/src/tests.rs [hunk 1]
```diff
@@ -495,3 +495,382 @@ fn asset_vc_status_change_should_succeed() {
 		));
 	});
 }
+#[test]
+fn test_asset_status_in_same_state_1() {
```
Comment (vatsa287) [inline pallets/asset/src/tests.rs]:
```rust
fn changing_status_of_asset_instance_with_same_status_should_fail()

```

File: pallets/asset/src/tests.rs [hunk 2]
```diff
@@ -495,3 +495,382 @@ fn asset_vc_status_change_should_succeed() {
...
+				Some(instance_id.clone()),
+				AssetStatusOf::ACTIVE
+			),
+			Error::<Test>::AssetInSameState
+		);
+	});
+}
+
+#[test]
+fn test_asset_status_in_same_state_2() {
```
Comment (vatsa287) [inline pallets/asset/src/tests.rs]:
```rust
fn changing_status_of_asset_with_same_status_should_fail()

```

File: pallets/asset/src/tests.rs [hunk 3]
```diff
@@ -495,3 +495,382 @@ fn asset_vc_status_change_should_succeed() {
...
+				Some(instance_id),
+				AssetStatusOf::ACTIVE,
+			),
+			Error::<Test>::AssetInSameState
+		);
+	});
+}
+
+#[test]
+fn test_vc_asset_status_in_same_state_1() {
```
Comment (vatsa287) [inline pallets/asset/src/tests.rs]:
```rust
fn changing_status_of_vc_asset_instance_with_same_status_should_fail()

```

File: pallets/asset/src/tests.rs [hunk 4]
```diff
@@ -495,3 +495,382 @@ fn asset_vc_status_change_should_succeed() {
...
+				None,
+				AssetStatusOf::ACTIVE,
+			),
+			Error::<Test>::AssetInSameState
+		);
+	});
+}
+
+#[test]
+fn test_vc_asset_status_in_same_state_2() {
```
Comment (vatsa287) [inline pallets/asset/src/tests.rs]:
```rust
fn changing_status_of_asset_with_same_status_should_fail()

```

**Review by vatsa287 — CHANGES_REQUESTED** [review]
Few minor comments

**Comment by zeel991** [comment]
Hey @vatsa287 , I have made the changes , Please review the PR

**Review by vatsa287 — APPROVED** [review]
LGTM.
Hoping you have gone through this [discussion](https://github.com/dhiway/cord/discussions/396), please start contributing on next level issues other than `good-first-issues`.

**Comment by zeel991** [comment]
Hello @amarts , can you please review the PR

**Comment by zeel991** [comment]
Hello @amarts , can you please review the PR

**Comment by smohan-dw** [comment]
LGTM. @zeel991 need to resolve the merge conflicts.

**Comment by zeel991** [comment]
> LGTM. @zeel991 need to resolve the merge conflicts.

Done !

**Comment by zeel991** [comment]
Hello @amarts sir ,PTAL

**Review by amarts — APPROVED** [review]

**Pull request merged**